packages/annotation-toolkit/src/panels/ContentPanel.js

```diff
@@ -0,0 +1,147 @@
+import React, { useContext } from "react";
+import { useStore } from "global-context-store";
+
+import { Menu, MenuDivider, Classes } from "@blueprintjs/core";
+
+function isFunction(functionToCheck) {
+  return (
+    functionToCheck && {}.toString.call(functionToCheck) === "[object Function]"
+  );
+}
+
+function ContentPanel() {
+  const store = useStore();
+  const { state, get } = store;
+
+  const isGroup = (layer, otherLayer) => {
+    if (!layer || !otherLayer) return false;
+    const parent = layer.slice(0, -1);
+    const otherParent = otherLayer.slice(0, -1);
+    if (otherLayer.join("|") === parent.join("|")) return true;
+    if (otherParent.join("|") === parent.join("|")) return true;
+    return false;
+  };
+
+  const layers = get(["layers"]);
+  if (!layers) return null;
+  const alwaysOnLayers = Object.values(layers).filter((layer) => {
+    return (
+      layer.alwaysOn === true ||
+      (isFunction(layer.alwaysOn) && layer.alwaysOn())
+    );
+  });
+  const groupedLayers = Object.values(layers).filter((layer) => {
+    return (
+      layer.onWithGroup === true &&
+      isGroup(layer.id.split("|"), state.selectedLayer)
+    );
+  });
+
+  let Noop = () => null;
+
+  let SelectedViewComponent = Noop;
+  const selectedViewName = state.selectedLayer || null;
+  let selectedLayer;
+  if (selectedViewName) {
+    const key = selectedViewName.join("|");
+    selectedLayer = get(["layers", key]);
+    if (
+      selectedLayer?.component &&
+      !selectedLayer.alwaysOn &&
+      groupedLayers.indexOf(selectedLayer) < 0
+    ) {
+      SelectedViewComponent = selectedLayer.component;
+    }
+  }
+
+  function gatherActions() {
+    if (!state.selectedLayer) return { actions: [], path: [] };
+    return state.selectedLayer.reduce(
+      (acc, value) => {
+        const path = [...acc.path, value];
+        const component = get(["layers", path.join("|")]);
+        if (!component) return acc;
+        const actions = component.actions
+          ? [...acc.actions, component.actions]
+          : acc.actions;
+        const actionPaths = component.actions
+          ? [...acc.actionPaths, path.join(" / ")]
+          : acc.actionPaths;
+        return { actions, path, actionPaths };
+      },
+      { actions: [], path: [], actionPaths: [] }
+    );
+  }
```

Comment:
This method gathers all the actions for the currently active layer as well as all of its parents up the tree.